cmd/juju/cloud/add.go

```diff
@@ -20,6 +20,14 @@ import (
 	"github.com/juju/juju/environs"
 )
 
+type CloudQueryer interface {
+	ParseCloudMetadataFile(path string) (map[string]cloud.Cloud, error)
```

Comment:
From a product standpoint, yes, we need to know that it is doing the correct thing. This is the responsibility of functional and/or CI tests. The responsibility of these tests is only to know that it's gluing the invocation of commands together with the `CloudMetadataStore` code correctly.

We can be reasonably sure that it's doing the correct thing because of the emergent correctness between these tests and [these](https://github.com/kat-co/juju/blob/interactive-bootstrap-cloud-name/cloud/clouds_test.go) tests.

This is why we pass in a well-tested, orthogonal module: so that this code, and these tests, don't have to re-test the same functionality. Functional tests will eclipse both of these packages, and this is why you have fewer functional tests (see [this](http://martinfowler.com/bliki/TestPyramid.html)).